luxonis_ml/nn_archive/config_building_blocks/base_models/input.py

```diff
@@ -65,6 +68,42 @@ class Input(CustomBaseModel):
         max_length=5,
         description="Shape of the input data as a list of integers (e.g. [H,W], [H,W,C], [BS,H,W,C], ...).",
     )
+    layout: str = Field(
+        description="Lettercode interpretation of the input layout (e.g., 'HCHW').",
+        default="NCHW",
+    )
     preprocessing: PreprocessingBlock = Field(
         description="Preprocessing steps applied to the input data."
     )
+
+    @model_validator(mode="before")
+    def validate_layout(
+        cls,
+        values,
+    ):
+        if "layout" in values.keys():
+            values["layout"] = values["layout"].upper()
+
+            if len(values["layout"]) != len(values["shape"]):
+                raise ValueError("Layout and shape must have the same length.")
+
+            if values["layout"][0] != "N":
```

Comment:
Not sure if we want to always enforce it, but might be good to just check that if it is present, it'd be in the first place?

Reply:
Fixed in [fdd6bff](https://github.com/luxonis/luxonis-ml/pull/109/commits/fdd6bff9eee194d210a767e258c1e47cd2f25986). Thanks!